File: Practice/18/Python/18.py
words = ("hallo", "klempner", "das", "ist", "fantastisch", "fluggegecheimen")
length = 0
probability = 1
for word in words:
	length += len(word)
userinput = input()
for userletter in userinput:
	variants = 0
	for word in words:
		for letter in word:
			if userletter == letter:
				variants+=1
	probability *= variants / length
print(probability)

File: Practice/18/C++/18/18.cpp
#include "pch.h"
#include <iostream>
#include <string>

int main()
{
	std::string words[]{"hallo", "klempner", "das", "ist", "fantastisch", "fluggegecheimen"};
	int length = 0;
	double probability(1);
	for (const std::string &word : words)
	{
		length += word.size();
	}
	while(true) 
	{
		char input = std::cin.get();
		if (input == '\n')
			break;
		int variants(0);
		for (const std::string &word : words)
		{
			for (const char &letter : word) 
			{
				if (input == letter)
					variants++;
			}
		}
	probability = (probability * variants) / length;
	}
	printf("%g",probability);
}
